### PR TITLE
fix: Support for computing series across multiple record batches

### PR DESCRIPTION
**Pull request by alamb**
__NOTE__: A whitespace blind diff makes it easier (for me at least) to see what is actually changed in this PR: https://github.com/influxdata/influxdb_iox/pull/4444/files?w=1

# Rationale
Closes https://github.com/influxdata/influxdb_iox/issues/4401

We are hitting the `"Computing series across multiple record batches not yet supported"` error as we roll IOx out to production

# Changes
1. Implement support by concatenating all record batches together.
2. https://github.com/influxdata/influxdb_iox/issues/4445  tracks improving this code to be streaming

cc @onelson and @crepererum

### CODE REVIEW

File: query/src/exec/seriesset/converter.rs [hunk 1]
```diff
@@ -83,59 +90,62 @@ impl SeriesSetConverter {
...
-            // emit each series
-            let series_sets = intersections.iter().map(|end_row| {
+        // for now, this logic only handles a single `RecordBatch` so
+        // concat data together.
+        //
+        // proper streaming support is tracked by: XXX
+        let batches = collect(it).await.context(ReadingSnafu)?;
+
+        let batch = if !batches.is_empty() {
+            RecordBatch::concat(&batches[0].schema(), &batches).context(ConcatenatingSnafu)?
```
Comment (alamb) [inline query/src/exec/seriesset/converter.rs:103]: this is the core change. The rest of the PR is testing

**Review by crepererum — APPROVED** [review]
Props for the test setup :muscle:

File: query/src/exec/seriesset/converter.rs [hunk 2]
```diff
@@ -508,237 +518,229 @@ mod tests {
...
+            let series_set1 = &results[0];
+
+            assert_eq!(series_set1.table_name.as_ref(), "foo");
+            assert_eq!(series_set1.tags, str_pair_vec_to_vec(&[("tag_a", "one")]));
+            assert_eq!(
+                series_set1.field_indexes,
+                FieldIndexes::from_timestamp_and_value_indexes(4, &[3])
+            );
+            assert_eq!(series_set1.start_row, 0);
+            assert_eq!(series_set1.num_rows, 3);
```
Comment (NGA-TRAN) [inline query/src/exec/seriesset/converter.rs:674]: 👍

File: query/src/exec/seriesset/converter.rs [hunk 3]
```diff
@@ -508,237 +518,229 @@ mod tests {
...
+            let series_set2 = &results[1];
+
+            assert_eq!(series_set2.table_name.as_ref(), "foo");
+            assert_eq!(series_set2.tags, str_pair_vec_to_vec(&[("tag_a", "two")]));
+            assert_eq!(
+                series_set2.field_indexes,
+                FieldIndexes::from_timestamp_and_value_indexes(4, &[3])
+            );
+            assert_eq!(series_set2.start_row, 3);
+            assert_eq!(series_set2.num_rows, 2);
```
Comment (NGA-TRAN) [inline query/src/exec/seriesset/converter.rs:685]: 👍

**Review by NGA-TRAN — COMMENTED** [review]
LGTM

**Pull request merged**